### PR TITLE
restore CSE'd node metadata in runtime asserts pass

### PR DESCRIPTION
**Pull request by pianpwk**
Adds val, and optionally stack_trace & nn_module_stack metadata back to SymInt compute nodes that we CSE, with a hook on `graph.create_node()`. Not sure if there's other metadata we want to populate here?

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

File: torch/fx/passes/runtime_assert.py [hunk 1]
```diff
@@ -145,6 +147,24 @@ def _is_intermediate_tensor_sym_call(node: fx.Node) -> bool:
             )
         )
 
+    # Figure out if we're in Dynamo or AOTAutograd
+    val_key = "val" if export else "example_value"
```
Comment (pianpwk) [inline torch/fx/passes/runtime_assert.py:151]: Not sure what the best way is to figure out which key to use. Maybe we could iterate through nodes and check what's present? And then fallback to this flag if that's empty for some reason.

Comment (ezyang) [reply torch/fx/passes/runtime_assert.py:151]: I usually just make a helper that fetches either val or example_value, depending on which one is set, preferring val. There's one in symbolic_shapes

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
This PR regressed instruction count on update_hint_regression from 10.59923 B to 10.72784 B.

image

The recovery in this chart was caused by an unrelated optimization https://github.com/pytorch/pytorch/pull/135256

**Pull request closed without merging**